### ISSUE TITLE
Allow Bundler/GemComment to accept comments for a gem group.

### ISSUE THREAD

**Issue by ThomasOwens** [rubocop/rubocop#12039]
## Is your feature request related to a problem? Please describe.

I like the idea behind Bundler/GemComment, but it has a lot of what I consider false positives. Rubocop is a good example, since there are multiple extensions gems. It doesn't seem appropriate to have a comment for each one, since it's possible to have a comment that describes the group of gems, their relationship, and any specific restrictions related to that group.

## Describe the solution you'd like

This would be considered good, as each gem group has a comment:

```
# See https://guides.rubyonrails.org/debugging_rails_applications.html#debugging-with-the-debug-gem
gem 'debug', platforms: %i[mri mingw x64_mingw]

# Linting and formatting
gem 'rubocop', require: false
gem 'rubocop-minitest', require: false
gem 'rubocop-performance', require: false
gem 'rubocop-rails', require: false
gem 'rubocop-rake', require: false

# Security
gem 'brakeman', require: false
```

This would be considered bad, as the brakeman gem doesn't have a comment:

```
# See https://guides.rubyonrails.org/debugging_rails_applications.html#debugging-with-the-debug-gem
gem 'debug', platforms: %i[mri mingw x64_mingw]

# Linting and formatting
gem 'rubocop', require: false
gem 'rubocop-minitest', require: false
gem 'rubocop-performance', require: false
gem 'rubocop-rails', require: false
gem 'rubocop-rake', require: false

gem 'brakeman', require: false
```

## Describe alternatives you've considered

It seems like the only alternative is disabling the rule. Although for an open source project, it is one less automated check that can be done if someone introduces a new gem.
